File: go/media/player.go
package main

import (
	"fmt"
	"net"
)

var playerState = "stopped"

func handleCommand(cmd string, params map[string]interface{}) {
	switch cmd {
	case "play":
		playerState = "playing"
		fmt.Println("Player: 播放")
	case "pause":
		playerState = "paused"
		fmt.Println("Player: 暂停")
	case "stop":
		playerState = "stopped"
		fmt.Println("Player: 停止")
	default:
		fmt.Println("Player: 未知命令", cmd)
	}
	fmt.Println("当前状态:", playerState)
}

func main() {
	ln, err := net.Listen("tcp", ":9100")
	if err != nil {
		panic(err)
	}
	defer ln.Close()
	fmt.Println("Player listening on 127.0.0.1:9100")
	conn, err := ln.Accept()
	if err != nil {
		panic(err)
	}
	defer conn.Close()
	buf := make([]byte, 1024)
	for {
		n, err := conn.Read(buf)
		if err != nil {
			break
		}
		cmd, params, err := UnpackCommand(buf[:n])
		if err != nil {
			fmt.Println("Unpack error:", err)
			continue
		}
		handleCommand(cmd, params)
	}
}

File: py/media/player.py
import socket
from media_protocol import MediaControlProtocol

HOST = '127.0.0.1'
PORT = 9100

player_state = 'stopped'

def handle_command(cmd, params):
    global player_state
    if cmd == 'play':
        player_state = 'playing'
        print('Player: 播放')
    elif cmd == 'pause':
        player_state = 'paused'
        print('Player: 暂停')
    elif cmd == 'stop':
        player_state = 'stopped'
        print('Player: 停止')
    else:
        print(f'Player: 未知命令 {cmd}')
    print(f'当前状态: {player_state}')

if __name__ == '__main__':
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.bind((HOST, PORT))
        s.listen(1)
        print(f'Player listening on {HOST}:{PORT}')
        conn, addr = s.accept()
        with conn:
            print('Connected by', addr)
            while True:
                data = conn.recv(1024)
                if not data:
                    break
                cmd, params = MediaControlProtocol.unpack_command(data)
                handle_command(cmd, params) 